src/Main.elm

```diff
@@ -106,6 +131,26 @@ update msg model =
                 Err _ ->
                     ( { model | phrases = Failure }, Cmd.none )
 
+        WordInput input ->
+            let
+                ( debouncer, cmd ) =
+                    debounce input model.debouncer
+            in
+            ( { model | debouncer = debouncer, wordInput = input, rhymes = NotAsked }, cmd )
+
+        FinishedWord id ->
```

Comment:
There's some fairly tight coupling between the debouncer, this action, and id verification. I haven't looked at the more generalized version of the debouncer, but within this commit, I might consider extracting some of this apart to be slightly more typesafe and specific to the debouncer.

Specifically, passing a primitive `Int` around, id comparison in here, and in `debounce` below, id incrementing, all seem slightly fragile in that it's not clear how they're tied together and how they interact.